<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::ApplicationModal</enum>
  </property>
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>800</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>openocd-svd</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="documentMode">
   <bool>false</bool>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <property name="unifiedTitleAndToolBarOnMac">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
      <property name="spacing">
       <number>6</number>
      </property>
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>6</number>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="btn_connect">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="lab_status">
          <property name="text">
           <string>Not connected to OpenOCD</string>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByMouse</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combo_periph">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentText">
           <string>No SVD file</string>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <item>
           <property name="text">
            <string>No SVD file</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTabWidget" name="tab_periph">
        <property name="tabPosition">
         <enum>QTabWidget::South</enum>
        </property>
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="tabsClosable">
         <bool>true</bool>
        </property>
        <property name="movable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>600</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="act_open_svd"/>
    <addaction name="separator"/>
    <addaction name="act_quit"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="act_autoread"/>
    <addaction name="act_autowrite"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="act_about"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSettings"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="act_quit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit the utility</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="act_autoread">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Auto read</string>
   </property>
   <property name="statusTip">
    <string>Control registers automatic reading</string>
   </property>
  </action>
  <action name="act_autowrite">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Auto write</string>
   </property>
   <property name="statusTip">
    <string>Control registers automatic writing</string>
   </property>
  </action>
  <action name="act_open_svd">
   <property name="text">
    <string>Open SVD file</string>
   </property>
   <property name="statusTip">
    <string>Open SVD file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="act_about">
   <property name="text">
    <string>About</string>
   </property>
   <property name="statusTip">
    <string>Show information about utility</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>act_quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>200</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_connect</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>btn_connect_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>50</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>act_open_svd</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>act_open_svd_triggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>act_autoread</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>act_autoread_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>act_autowrite</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>act_autowrite_triggered(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>combo_periph</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>combo_periph_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>act_about</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>act_about_triggered()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tab_periph</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>tab_periph_close(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>btn_connect_clicked()</slot>
  <slot>act_open_svd_triggered()</slot>
  <slot>act_autoread_triggered(bool)</slot>
  <slot>act_autowrite_triggered(bool)</slot>
  <slot>combo_periph_changed(int)</slot>
  <slot>act_about_triggered()</slot>
  <slot>tab_periph_close(int)</slot>
 </slots>
</ui>
